### PR TITLE
feat(tools): Add a tool to generate release notes docs from yaml files of release notes

### LINKED ISSUE #2273: Add tools and make target to generate `releases` docs via `release-notes`YAMLs
*Description*:

Add tools and make target to generate `site/content/en/latest/releases` docs via yamls in `release-notes`.

### PR DESCRIPTION
**Pull request by lemonlinger**
**What type of PR is this?**

**What this PR does / why we need it**:

Introduce a tool that creates release notes docs from YAML files of release notes, along with a make target to run it.

**Which issue(s) this PR fixes**:

Fixes #2273

### CODE REVIEW

File: site/content/en/latest/releases/v0.1.0.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ---
 title: "v0.1.0"
-publishdate: 2022-05-16
```
Comment (zirain) [inline site/content/en/latest/releases/v0.1.0.md:3]: can you revert these changes?

File: tools/make/docs.mk [hunk 2]
```diff
@@ -65,7 +65,7 @@ docs-release-prepare:
 	echo $(TAG) > VERSION
 
 .PHONY: docs-release-gen
-docs-release-gen:
+docs-release-gen:
```
Comment (zirain) [inline tools/make/docs.mk:68]: nit?

File: tools/src/release-notes-docs/yml2md.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    print("Markdown file '{}' has been generated.".format(output_markdown_file))
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print("Usage: python yml2md.py <input_yaml_file> <output_markdown_path>")
+        sys.exit(1)
+
+    input_yaml_file = sys.argv[1]
+    output_markdown_path = sys.argv[2]
+    convert_yaml_to_markdown(input_yaml_file, output_markdown_path)
```
Comment (zirain) [inline tools/src/release-notes-docs/yml2md.py:67]: nit: new line at the end

Comment (lemonlinger) [reply tools/src/release-notes-docs/yml2md.py]: Fix

**Review by zirain — APPROVED** [review]
LGTM, cc @Xunzhuo

**Review by Xunzhuo — COMMENTED** [review]
Why change v0.2.0-rc.1.md to  v0.2.0-rc1.md ?

**Comment by lemonlinger** [comment]
> Why change v0.2.0-rc.1.md to  v0.2.0-rc1.md ?

To make it consistent with its yaml filename.
